packages/runtime/container-runtime/src/garbageCollection.ts

```diff
@@ -826,13 +841,36 @@ export class GarbageCollector implements IGarbageCollector {
      * If these nodes are currently unreferenced, they will be assigned new unreferenced state by the current run.
      */
     private updateStateSinceLastRun(currentGCData: IGarbageCollectionData) {
-        // If we haven't run GC before or no references were added since the last run, there is nothing to do.
-        if (this.gcDataFromLastRun === undefined || this.referencesSinceLastRun.size === 0) {
+        // If we haven't run GC before there is nothing to do.
+        if (this.previousGCDataFromLastRun === undefined) {
             return;
         }
 
-        // Validate that we have identified all references correctly.
-        this.validateReferenceCorrectness(currentGCData);
+        // Find any references that haven't been identified correctly.
+        const missingExplicitReferences = this.findMissingExplicitReferences(
+            currentGCData,
+            this.previousGCDataFromLastRun,
+            this.newReferencesSinceLastRun,
+        );
+
+        // The following log will be enabled once this issue is resolved:
+        // https://github.com/microsoft/FluidFramework/issues/8878.
+        if(this.mc.config.getBoolean(logUnknownOutboundReferencesKey) === true
+            && missingExplicitReferences.length > 0) {
+            missingExplicitReferences.forEach((missingExplicitReference) => {
+                const event: ITelemetryPerformanceEvent = {
+                    eventName: "gcUnknownOutboundReferences",
+                    gcNode: missingExplicitReference[0],
+                    gcRoutes: JSON.stringify(missingExplicitReference[1]),
+                };
+                this.mc.logger.sendPerformanceEvent(event);
+            });
+        }
+
+        // No references were added since the last run so we don't need to run garbage collection
```

Comment:
Instead of "we don't to run gc", this should say something like "we don't have to update reference states of any unreferenced nodes"